### PR TITLE
Support Regexp on `update_project_provisioning` action

### PR DESCRIPTION
**Pull request by giginet**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

https://docs.fastlane.tools/actions/#update_project_provisioning

According to the documentation, we can pass Regular Expression for `target_filter` or `build_configuration`.

https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/update_project_provisioning.rb#L110

However, we can pass as `String` only. Errors are raised when we passed as `Regexp` object.

```
[15:02:39]: You passed invalid parameters to 'update_project_provisioning'.
[15:02:39]: Check out the error below and available options by running `fastlane action update_project_provisioning`
+------------------+----------+
|        Lane Context         |
+------------------+----------+
| DEFAULT_PLATFORM | ios      |
| PLATFORM_NAME    | ios      |
| LANE_NAME        | ios beta |
+------------------+----------+
[15:02:39]: 'target_filter' value must be a String! Found Regexp instead.
```

### Description

Make `target_configuration` and `target_filter` are enable to accept `Regexp` object.
Of course, you can also pass `String`.

### CODE REVIEW

File: fastlane/spec/actions_specs/update_project_provisioning_spec.rb [hunk 1]
```diff
@@ -0,0 +1,104 @@
+describe Fastlane do
+  describe Fastlane::FastFile do
+    describe "Update Project Provisioning" do
+      let (:fixtures_path) { "./fastlane/spec/fixtures" }
+      let (:xcodeproj) { File.absolute_path(File.join(fixtures_path, 'xcodeproj', 'bundle.xcodeproj')) }
+      let (:profile_path) { File.absolute_path(File.join(fixtures_path, 'profiles', 'test.mobileprovision')) }
```
Comment (giginet) [inline fastlane/spec/actions_specs/update_project_provisioning_spec.rb]: This fixture is copied from `match/specs/fixtures`

File: fastlane/lib/fastlane/actions/update_project_provisioning.rb [hunk 2]
```diff
@@ -108,15 +108,23 @@ def self.available_options
...
                                        description: "Legacy option, use 'target_filter' instead",
                                        optional: true),
           FastlaneCore::ConfigItem.new(key: :build_configuration,
                                        env_name: "FL_PROJECT_PROVISIONING_PROFILE_BUILD_CONFIGURATION",
                                        description: "A filter for the build configuration name. Use a standard regex. Applied to all configurations if not specified",
-                                       optional: true),
+                                       optional: true,
+                                       is_string: false,
+                                       verify_block: proc do |value|
+                                         UI.user_error!("target_filter should be Regexp or String") unless [Regexp, String].any? { |type| value.kind_of?(type) }
```
Comment (nafu) [inline fastlane/lib/fastlane/actions/update_project_provisioning.rb]: build_configuration would be 💯

**Review by nafu — COMMENTED** [review]
Looks good to me 👍 

Error message should be fixed.

File: fastlane/lib/fastlane/actions/update_project_provisioning.rb (hunk 2 above)
Comment (giginet) [inline fastlane/lib/fastlane/actions/update_project_provisioning.rb]: oops 🙀

**Review by nafu — APPROVED** [review]
Looks awesome 🎉

File: fastlane/spec/actions_specs/update_project_provisioning_spec.rb [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+          it 'should be invalid' do
+            expect do
+              Fastlane::FastFile.new.parse("lane :test do
+                  update_project_provisioning ({
+                    xcodeproj: '#{xcodeproj}',
+                    profile: '#{profile_path}',
+                    build_configuration: 1
+                })
+              end").runner.execute(:test)
+            end.to raise_error
```
Comment (KrauseFx) [inline fastlane/spec/actions_specs/update_project_provisioning_spec.rb]: Please specify the error type for all the tests 👍

**Comment by KrauseFx** [comment]
This is awesome, so many tests 👍 I added one comment before it's ready to be merged

**Comment by KrauseFx** [comment]
This is awesome, so many tests 👍 I added one comment before it's ready to be merged

**Comment by giginet** [comment]
@KrauseFx @nafu Thanks your reviews 😄 

I'll merge this 🚀

**Pull request merged**